src/content/docs/Standard Library/index.mdx

```diff
@@ -334,31 +334,116 @@ moving the output pointer 1 or 2 steps.
 ## std::io
 
 ### String! readline(stream = io::stdin(), Allocator allocator = allocator::heap())
-Read from a stream (default is stdin) to the next "\n" or to the end of the stream, whatever comes first.
+Read a `String!` from a file stream. Defaults to standard input (stdin). Reads to the next newline or to the end of stream. `Readline` returns an optional, and must be handled accordingly.
 
+```c
+import std::io;
+
+fn void! main() {
+    String! name; 
+
+    name = io::readline();
+    if (catch excuse = name) {
+        return excuse?;
+    }
+
+    io::printfn("The Matrix has you... Follow the white rabbit. Knock, knock, %s.", name);
+}
+```
```

Comment:
```suggestion
```c3
import std::io;

fn void! main() 
{
    String! name = io::readline();
    if (catch excuse = name) 
    {
        return excuse?;
    }

    io::printfn("Name was: %s.", name);
}
```
```

trying to minimise line length as much as possible (so it works nicely on mobile screens) and newlines for `{` as per the style we've got for the rest of the guide